src/components/TextSize.js

```diff
@@ -1,32 +1,62 @@
 import { h } from 'preact'
-import { useI18n, useSoftkey, useArticleTextSize } from 'hooks'
+import { useEffect, useContext } from 'preact/hooks'
+import { useI18n, useSoftkey } from 'hooks'
 import { articleTextSize } from 'utils'
+import { FontContext } from 'contexts'
+
+let originalFontSizeClass
 
 export const TextSize = ({ close }) => {
   const i18n = useI18n()
 
-  const [textsize, setTextSize] = useArticleTextSize()
-  const { onKeyArrowLeft, onKeyArrowRight } = articleTextSize.getSoftkeyEffect(setTextSize)
+  const { fontSizeClass, setFontSizeClass } = useContext(FontContext)
+
+  useEffect(() => {
+    originalFontSizeClass = articleTextSize.getFontSizeClassName()
+  }, [])
+
+  const onKeyFixedArrowLeft = () => {
+    articleTextSize.adjust(-1)
+    setFontSizeClass(articleTextSize.getFontSizeClassName())
+  }
+
+  const onKeyFixedArrowRight = () => {
+    articleTextSize.adjust(1)
+    setFontSizeClass(articleTextSize.getFontSizeClassName())
+  }
+
+  const onKeyBackspace = () => {
+    const originalSizeInteger = parseTextSizeInteger(originalFontSizeClass)
+    const currentSizeInteger = parseTextSizeInteger(articleTextSize.getFontSizeClassName())
+
+    articleTextSize.adjust(originalSizeInteger - currentSizeInteger)
+    setFontSizeClass(articleTextSize.getFontSizeClassName())
+    close()
+  }
+
   const sliderValue = ['0', '16.6', '33.2', '49', '66.6', '83.2', '94']
+  const parseTextSizeInteger = (fontSizeClass) => {
+    return parseInt(fontSizeClass.slice(-1))
```

Comment:
Following an earlier comment, the class should be built as needed based on the font size, the font size should not have to be extracted from the class.